blogs/parser.py

```diff
@@ -37,13 +38,20 @@ def _render_blog_supernav(entry):
 
 def update_blog_supernav():
     """Retrieve latest entry and update blog supernav item """
-    latest_entry = BlogEntry.objects.filter(feed_id=1).latest()
-    rendered_box = _render_blog_supernav(latest_entry)
-
-    box, _ = Box.objects.update_or_create(
-        label='supernav-python-blog',
-        defaults={
-            'content': rendered_box,
-            'content_markup_type': 'html',
-        }
-    )
+    try:
+        latest_entry = BlogEntry.objects.filter(
+            feed=Feed.objects.get(
+                feed_url=settings.PYTHON_BLOG_FEED_URL
+            )
+        ).latest()
+    except (BlogEntry.DoesNotExist, Feed.DoesNotExist):
+        pass
+    else:
+        rendered_box = _render_blog_supernav(latest_entry)
+        box, _ = Box.objects.update_or_create(
+            label='supernav-python-blog',
+            defaults={
+                'content': rendered_box,
+                'content_markup_type': 'html',
+            }
+        )
```

Comment:
This change makes it not rely on a hardcoded primary key and it doesn't crash when the feed doesn't exist or there are no BlogEntries at all. Which will happen on a "fresh" installation.

However, I'm not sure if the 'supernav-python-blog' is actually used anymore. I don't see it on the live site. So maybe the entire supernav stuff can be removed?

Reply:
'supernav-python-blog' is used in `static/js/script.js`.

Your rewrite is much better than the original code.